kolibri/core/public/api.py

```diff
@@ -146,3 +156,90 @@ def get_public_file_checksums(request, version):
         json.dumps({"id": error_constants.NOT_FOUND, "metadata": {"view": ""}}),
         content_type="application/json",
     )
+
+
+def position_in_queue(id):
+    qs = SyncQueue.objects.all().order_by("datetime").values("id")
+    pos = 0  # in case the queue is empty
+    for pos, value in enumerate(qs):
+        if value["id"] == id:
+            break
+    return pos
+
+
+class SyncQueueViewSet(viewsets.ViewSet):
+    def list(self, request):
+        """Returns length of the queue for each of the available facilities"""
+        SyncQueue.clean_stale()  # first, ensure not expired devices are in the queue
+        facilities = Facility.objects.all()
+        queue = {}
+        for facility in facilities:
+            queue[facility.id] = SyncQueue.objects.filter(
+                user__facility=facility
+            ).count()
+        return Response(queue)
+
+    def check_queue(self, pk=None):
+        current_transfers = TransferSession.objects.filter(active=True).count()
+        if current_transfers <= MAX_CONCURRENT_SYNCS:
+            allow_sync = True
+            data = {"action": SYNC}
+        else:
+            polling = current_transfers + SyncQueue.objects.all().count()
```

Comment:
Should we only do this if `pk is None`?

Reply:
no, it should be done always. If pk is None , it's creating the element in the queue and we should provide a minimal keep_alive time. Originally it would be the length of the queue + numbers of syncs in progress, but after the conversation we had last Monday when we decided to enqueue if there's one sync in progress it should be something like 
`polling = minimal_ttl + SyncQueue.objects.all().count()`

Reply:
Oh, my thought was that in the case that the `pk is not None` we throw this value away inside the `if` statement below, so there's no need to call this count unless the `pk is None`.

Reply:
Ahh, I understood what you meant, done